cache/cache.cpp

```diff
@@ -190,11 +194,332 @@ size_t cache_manager::idx(const unsigned char *id) {
 	return (i ^ j) % m_caches.size();
 }
 
+write_request::write_request(struct dnet_io_attr *io, char *data)
+: ioflags(io->flags)
+, user_flags(io->user_flags)
+, timestamp(io->timestamp)
+, json_size(0)
+, json_capacity(0)
+, data_offset(io->offset)
+, data_size(io->size)
+, data_capacity(0)
+, data_commit_size(0)
+, cache_lifetime(io->start)
+, data_checksum(&io->parent)
+, request_data(io)
+, data(data)
+, json(nullptr)
+{
+	dnet_empty_time(&json_timestamp);
+}
+
+write_request::write_request(struct ioremap::elliptics::dnet_write_request &req, void *request_data, char *data)
+: ioflags(req.ioflags)
+, user_flags(req.user_flags)
+, timestamp(req.timestamp)
+, json_size(req.json_size)
+, json_capacity(req.json_capacity)
+, json_timestamp(req.json_timestamp)
+, data_offset(req.data_offset)
+, data_size(req.data_size)
+, data_capacity(req.data_capacity)
+, data_commit_size(req.data_commit_size)
+, cache_lifetime(req.cache_lifetime)
+, data_checksum(nullptr)
+, request_data(request_data)
+, data(data + json_size)
+, json(json_size ? data : nullptr)
+{
+}
+
 }} /* namespace ioremap::cache */
 
 using namespace ioremap::cache;
 
-int dnet_cmd_cache_io(struct dnet_backend_io *backend, struct dnet_net_state *st, struct dnet_cmd *cmd, struct dnet_io_attr *io, char *data)
+static int dnet_cmd_cache_io_write(struct cache_manager *cache,
+                                   struct dnet_net_state *st,
+                                   struct dnet_cmd *cmd,
+                                   struct dnet_io_attr *io,
+                                   char *data)
+{
+	int err;
```

Comment:
Why doesn't it check `ioflags & DNET_IO_FLAGS_NOCACHE` like `dnet_cmd_cache_io_write_new` do?